lib/vintage_net_wizard/web/endpoint.ex

```diff
@@ -58,33 +64,33 @@ defmodule VintageNetWizard.Web.Endpoint do
     ]
   end
 
-  defp maybe_use_ssl(_use_ssl = true) do
-    ssl_dir = ssl_dir()
-
-    Plug.Cowboy.child_spec(
-      plug: Router,
-      scheme: :https,
-      options: [
-        dispatch: dispatch(),
-        certfile: Application.get_env(:vintage_net_wizard, :certfile, "#{ssl_dir}/cert.pem"),
-        keyfile: Application.get_env(:vintage_net_wizard, :keyfile, "#{ssl_dir}/key.pem"),
-        port: Application.get_env(:vintage_net_wizard, :port, 443)
-      ]
-    )
-  end
-
-  defp maybe_use_ssl(_no_ssl) do
-    Plug.Cowboy.child_spec(
-      plug: Router,
-      scheme: :http,
-      options: [
-        dispatch: dispatch(),
-        port: Application.get_env(:vintage_net_wizard, :port, 80)
-      ]
-    )
+  defp maybe_use_ssl(_use_ssl = true, opts) do
+    with certfile <- Keyword.get(opts, :certfile, {:error, :no_certfile}),
```

Comment:
Going with the idea of the `ssl: []` key, maybe we shouldn't try to manually handle specific ssl options since the `Plug.Cowboy.child_spec/1` passes the options to `Plug.SSL` which [accepts and validates all erlang `:ssl` module options](https://hexdocs.pm/plug/1.7.0/Plug.SSL.html#configure/1-options), so we can let the validation happen there.

That would also allow passing `:cert` and `:key` values instead of a file path to be used.

Thoughts?